buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/YamlRestCompatTestPlugin.java

```diff
@@ -73,84 +74,54 @@ public void apply(Project project) {
-            .getBuildDir()
-            .toPath()
-            .resolve("bwc")
-            .resolve("checkout-" + priorMajorVersion + ".x");
-
         // copy compatible rest specs
-        Configuration compatSpec = project.getConfigurations().create("compatSpec");
-        Configuration xpackCompatSpec = project.getConfigurations().create("xpackCompatSpec");
-        Configuration additionalCompatSpec = project.getConfigurations().create("additionalCompatSpec");
+        Configuration bwcMinorConfig = project.getConfigurations().create("bwcMinor");
+        Dependency bwcMinor = project.getDependencies().project(Map.of("path", ":distribution:bwc:minor", "configuration", "checkout"));
+
         Provider<CopyRestApiTask> copyCompatYamlSpecTask = project.getTasks()
             .register("copyRestApiCompatSpecsTask", CopyRestApiTask.class, task -> {
+                project.getDependencies().add(bwcMinorConfig.getName(), bwcMinor);
+                task.dependsOn(bwcMinorConfig);
+                task.coreConfig = bwcMinorConfig;
+                task.xpackConfig = bwcMinorConfig;
+                task.additionalConfig = bwcMinorConfig;
                 task.includeCore.set(extension.restApi.getIncludeCore());
                 task.includeXpack.set(extension.restApi.getIncludeXpack());
                 task.sourceSetName = SOURCE_SET_NAME;
                 task.skipHasRestTestCheck = true;
-                task.coreConfig = compatSpec;
-                project.getDependencies()
-                    .add(
-                        task.coreConfig.getName(),
-                        project.files(checkoutDir.resolve("rest-api-spec/src/main/resources").resolve(RELATIVE_API_PATH))
-                    );
-                task.xpackConfig = xpackCompatSpec;
-                project.getDependencies()
-                    .add(
-                        task.xpackConfig.getName(),
-                        project.files(checkoutDir.resolve("x-pack/plugin/src/test/resources").resolve(RELATIVE_API_PATH))
-                    );
-                task.additionalConfig = additionalCompatSpec;
-                // per project can define custom specifications
-                project.getDependencies()
-                    .add(
-                        task.additionalConfig.getName(),
-                        project.files(
-                            getCompatProjectPath(project, checkoutDir).resolve("src/yamlRestTest/resources").resolve(RELATIVE_API_PATH)
-                        )
-                    );
-                task.dependsOn(task.coreConfig);
-                task.dependsOn(task.xpackConfig);
-                task.dependsOn(task.additionalConfig);
-                task.dependsOn(":distribution:bwc:minor:checkoutBwcBranch");
+                task.coreConfigToFileTree = config -> project.fileTree(
+                    bwcMinorConfig.getSingleFile().toPath().resolve("rest-api-spec/src/main/resources").resolve(RELATIVE_API_PATH)
```

Comment:
shouldn't we really be referencing `config` here instead of `bwcMinorConfig` even though they are almost certain to be the same?